hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/datanode/container/ChecksumSubcommand.java

```diff
@@ -0,0 +1,229 @@
+  }
+
+  /**
+   * Reads the container checksum info from the specified file.
+   */
+  private ContainerProtos.ContainerChecksumInfo readChecksumInfo(File treeFile) throws IOException {
+    try (InputStream inputStream = Files.newInputStream(treeFile.toPath())) {
+      return ContainerProtos.ContainerChecksumInfo.parseFrom(inputStream);
+    }
+  }
+
+  /**
+   * Wrapper class for JSON serialization of container checksum info.
+   */
+  private static class ChecksumInfoWrapper {
+    private final long containerID;
+    private final String filePath;
+    private final ContainerMerkleTreeWrapper containerMerkleTree;
+
+    ChecksumInfoWrapper(ContainerProtos.ContainerChecksumInfo checksumInfo, String filePath) {
+      this.containerID = checksumInfo.getContainerID();
+      this.filePath = filePath;
+      this.containerMerkleTree = checksumInfo.hasContainerMerkleTree() ? 
+          new ContainerMerkleTreeWrapper(checksumInfo.getContainerMerkleTree()) : null;
+    }
+
+    public long getContainerID() {
+      return containerID;
+    }
+
+    public String getFilePath() {
+      return filePath;
+    }
+
+    public ContainerMerkleTreeWrapper getContainerMerkleTree() {
+      return containerMerkleTree;
+    }
+  }
+
+  /**
+   * Wrapper class for JSON serialization of container merkle tree.
+   */
+  private static class ContainerMerkleTreeWrapper {
+    @JsonSerialize(using = JsonUtils.ChecksumSerializer.class)
+    private final long dataChecksum;
+    private final List<BlockMerkleTreeWrapper> blockMerkleTrees;
+
+    ContainerMerkleTreeWrapper(ContainerProtos.ContainerMerkleTree merkleTree) {
+      this.dataChecksum = merkleTree.hasDataChecksum() ? merkleTree.getDataChecksum() : 0L;
+      this.blockMerkleTrees = new ArrayList<>();
+      for (ContainerProtos.BlockMerkleTree blockTree : merkleTree.getBlockMerkleTreeList()) {
+        this.blockMerkleTrees.add(new BlockMerkleTreeWrapper(blockTree));
+      }
+    }
+
+    public long getDataChecksum() {
+      return dataChecksum;
+    }
+
+    public List<BlockMerkleTreeWrapper> getBlockMerkleTrees() {
+      return blockMerkleTrees;
+    }
+  }
+
+  /**
+   * Wrapper class for JSON serialization of block merkle tree.
+   */
+  private static class BlockMerkleTreeWrapper {
+    private final long blockID;
+    private final boolean deleted;
+    @JsonSerialize(using = JsonUtils.ChecksumSerializer.class)
+    private final long dataChecksum;
+    private final List<ChunkMerkleTreeWrapper> chunkMerkleTrees;
+
+    BlockMerkleTreeWrapper(ContainerProtos.BlockMerkleTree blockTree) {
+      this.blockID = blockTree.getBlockID();
+      this.deleted = blockTree.getDeleted();
+      this.dataChecksum = blockTree.hasDataChecksum() ? blockTree.getDataChecksum() : 0L;
+      this.chunkMerkleTrees = new ArrayList<>();
+      
+      // Only include chunk trees if block is not deleted
+      if (!deleted) {
```

Comment:
I think we should literally print whatever is in the file since this is a debug tool. Chunks should be removed when the file is written if the block is deleted but it would be good if this tool could be used to verify that.